internal/integratedservices/services/securityscan/operator.go

```diff
@@ -187,6 +175,42 @@ func (op IntegratedServiceOperator) Apply(ctx context.Context, clusterID uint, s
 	return nil
 }
 
+func (op IntegratedServiceOperator) apply(ctx context.Context, registrySpec *registrySpec, anchoreClient anchore.AnchoreClient) error {
+	secret, err := op.secretStore.GetSecretValues(ctx, registrySpec.SecretID)
+	if err != nil {
+		return errors.WrapWithDetails(err, "failed to get anchore registry secret", "secretId", registrySpec.SecretID, "registry", registrySpec.Registry)
+	}
+
+	registry := anchore.Registry{
+		Type:     registrySpec.Type,
+		Registry: registrySpec.Registry,
+		Verify:   !registrySpec.Insecure,
+	}
+
+	if anchore.IsEcrRegistry(registrySpec.Registry) {
+		registry.Username = secret[secrettype.AwsAccessKeyId]
+		registry.Password = secret[secrettype.AwsSecretAccessKey]
+	} else {
+		registry.Username = secret[secrettype.Username]
+		registry.Password = secret[secrettype.Password]
+	}
```

Comment:
The secret type should be validated before using the secret here. I know this is not new code, but I couldn't find any validation, so please open an issue.

Reply:
https://github.com/banzaicloud/pipeline/issues/3404